packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -267,7 +267,8 @@ class PlatformAssetBundle extends CachingAssetBundle {
       return ui.ImmutableBuffer.fromUint8List(bytes.buffer.asUint8List());
     }
     try {
-      return await ui.ImmutableBuffer.fromAsset(key);
+      final String encodedKey = Uri(path: Uri.encodeFull(key)).path;
```

Comment:
Technically we _can_ load an asset with a space in the name just fine - but historically we couldn't, so the tool has always done URI encoding when bundling. To match the current behavior, we need to perform the same encoding here

Reply:
We could remove the URI encoding from the tool, but that might get weird , and we may have other undiscovered bugs

Reply:
Is this the only call to `ImmutableBuffer.fromAsset()` that needs this fix? I'm also wondering if it's needed for `FragmentProgram.fromAsset()` which uses the asset key ~exactly the same way that `ImmutableBuffer` does.